crates/bevy_mesh/src/mesh.rs

```diff
@@ -389,19 +412,21 @@ impl Mesh {
     /// Retrieves the vertex `indices` of the mesh.
     #[inline]
     pub fn indices(&self) -> Option<&Indices> {
-        self.indices.as_ref()
+        self.indices.as_ref().expect("Mesh has been extracted to RenderWorld. To access vertex attributes, the mesh must have RenderAssetUsages::MAIN_WORLD").as_ref()
     }
 
     /// Retrieves the vertex `indices` of the mesh mutably.
     #[inline]
     pub fn indices_mut(&mut self) -> Option<&mut Indices> {
```

Comment:
All these mutation methods essentially stop functioning when the asset is extracted. What about runtime generated (and mutated) assets? Is the intent that users will like fully replace the Mesh with like `*mesh_mut = Mesh::new()`?

If so, this probably needs a migration guide.

Reply:
the pattern would be either to fully initialize in one step, or to initialize with `RenderAssetUsages::MAIN_WORLD`, make any modifications, then when it's finished change the `usage to RenderAssetUsages::RENDER_WORLD`. after that no more modifications are possible since the data is only on GPU and we can't get it back. 

if users want to make continuous runtime modifications, then they should use `MAIN_WORLD || RENDER_WORLD`, or fully replacing will also work.

this PR doesn't change anything in those scenarios, modifying an asset with `RENDER_WORLD` and not `MAIN_WORLD` is not possible currently either since the asset gets removed from the `Assets` collection. i can write something if you like, but the usage doesn't change.